beacon_node/http_api/src/lib.rs

```diff
@@ -1933,6 +1933,77 @@ pub fn serve<T: BeaconChainTypes>(
             })
         });
 
+    /*
+     * builder/states
+     */
+
+    let builder_states_path = eth_v1
+        .and(warp::path("builder"))
+        .and(warp::path("states"))
+        .and(chain_filter.clone());
+
+    // GET builder/states/{state_id}/expected_withdrawals
+    let get_expected_withdrawals = builder_states_path
+        .clone()
+        .and(warp::path::param::<StateId>())
+        .and(warp::query::<api_types::ExpectedWithdrawalsQuery>())
+        .and(warp::path("expected_withdrawals"))
+        .and(warp::path::end())
+        .and(warp::header::optional::<api_types::Accept>("accept"))
+        .and_then(
+            |chain: Arc<BeaconChain<T>>,
+             state_id: StateId,
+             query: api_types::ExpectedWithdrawalsQuery,
+             accept_header: Option<api_types::Accept>| {
+                blocking_response_task(move || {
+                    let proposal_slot = query.proposal_slot.unwrap_or(Slot::new(1));
```

Comment:
The default `proposal_slot` should be the slot after the pre state if it's  not specified. 

I think what you have works for the default case where `proposer_slot` is not specified (slot = `state.slot() + 1`), but if the current `state.slot` is 100503, and caller specifies a `proposer_slot` of 100504, it ends up adding both numbers up, and this gives you the incorrect proposal slot.

I think you may want something like:

```rust
let proposal_slot = query.proposal_slot.unwrap_or(state.slot() + 1);
```

Reply:
I think we should also validate that the `proposal_slot` is greater than `state.slot`, as we are computing the expected withdrawals based on the pre-state and slot number, which must be after the state.

Reply:
I wonder how far ahead of the prestate we should allow `proposal_slot` to be - if user provides a far future slot, we should probably return an error rather than computing it - although I haven't seen any specific number in the spec.

Reply:
> I wonder how far ahead of the prestate we should allow `proposal_slot` to be - if user provides a far future slot, we should probably return an error rather than computing it - although I haven't seen any specific number in the spec.

thats a good point. looking at the prysm [implementation](https://github.com/prysmaticlabs/prysm/pull/12519/files)  they calculated a look ahead limit using MaxSeedLookahead

`lookAheadLimit := uint64(params.BeaconConfig().SlotsPerEpoch.Mul(uint64(params.BeaconConfig().MaxSeedLookahead)))`

and then returned an error if `proposal slot > state.slot + lookAheadlimit`